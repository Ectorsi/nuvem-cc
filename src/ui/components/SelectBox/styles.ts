import styled from 'styled-components';
import { SelectProps } from './SelectBox';
import Theme from '../../../styles/theme';

const { colorError } = Theme.colors.main;

type SelectBoxStyledProps = Pick<SelectProps, 'error'>;

export const Wrapper = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: center;
`;

export const SelectContainer = styled.select<SelectBoxStyledProps>`
    border: ${({ error }) => (!!error ? `2px solid ${colorError}` : '1px solid #000')};
    border-radius: 4px;
    padding: 8px;
    width: 240px;

    @media (max-width: 519px) {
        width: 100%;
    }
`;

export const Label = styled.label`
    font-size: 14px;
    font-weight: 600;
    color: ${Theme.colors.font.secondary};
`;

export const Error = styled.span`
    color: ${Theme.colors.main.colorError};
`;
