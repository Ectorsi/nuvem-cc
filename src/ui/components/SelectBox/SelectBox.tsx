import { ChangeEvent } from 'react';
import * as S from './styles';

export type Option = {
    value: string;
    label: string;
};

export type SelectProps = {
    label: string;
    placeholder: string;
    value: string | undefined;
    onChange: (e: ChangeEvent<HTMLSelectElement>) => void;
    error?: string;
    options: Option[];
    disabled?: boolean;
};
const SelectBox = ({
    label,
    placeholder,
    error,
    value,
    onChange,
    options,
    disabled,
}: SelectProps) => {
    return (
        <S.Wrapper>
            <S.Label>{label}</S.Label>
            <S.SelectContainer
                disabled={disabled}
                value={value}
                onChange={onChange}
            >
                <option value="">{placeholder}</option>
                {options.map((option) => (
                    <option key={option.value} value={option.value}>
                        {option.label}
                    </option>
                ))}
            </S.SelectContainer>
            {!!error && <S.Error>{error}</S.Error>}
        </S.Wrapper>
    );
};

export default SelectBox;
