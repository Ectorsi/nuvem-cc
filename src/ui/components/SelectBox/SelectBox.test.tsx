import { render, screen } from '@testing-library/react';

import SelectBox, { SelectProps } from './SelectBox';

const mockSelectBox: SelectProps = {
    label: 'LabelTest',
    placeholder: 'PlaceholderTest',
    value: 'SelectrValue',
    onChange: jest.fn(),
    error: '',
    options: [
        { value: 'Option1', label: 'Option1' },
        { value: 'Option2', label: 'Option2' },
    ],
};

describe('SelectBox unit test', () => {
    it("should be able to show the SelectBox's label and input", () => {
        const { container } = render(<SelectBox {...mockSelectBox} />);
        const Label = screen.getAllByText(/Test/i)[0];
        const option = screen.getAllByText(/Option1/i)[0];
        expect(option).toBeInTheDocument();
        expect(Label).toBeInTheDocument();
        expect(container.firstChild).toMatchSnapshot();
    });

    it("should be able to show the SelectBox's error message", () => {
        render(<SelectBox {...mockSelectBox} error="error test" />);
        const errorMessage = screen.getByText(/error test/i);
        expect(errorMessage).toBeInTheDocument();
    });
});
