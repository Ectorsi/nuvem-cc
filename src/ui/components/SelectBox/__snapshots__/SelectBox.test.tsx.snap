// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectBox unit test should be able to show the SelectBox's label and input 1`] = `
<div
  class="sc-beqWaB gMwkTI"
>
  <label
    class="sc-dmqHEX hObCgG"
  >
    LabelTest
  </label>
  <select
    class="sc-gueYoa jRobxD"
  >
    <option
      value=""
    >
      PlaceholderTest
    </option>
    <option
      value="Option1"
    >
      Option1
    </option>
    <option
      value="Option2"
    >
      Option2
    </option>
  </select>
</div>
`;
